arkouda/sparrayclass.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import annotations
+
+import builtins
+from functools import reduce
+from typing import Optional, Sequence, Union
+
+import numpy as np
+from typeguard import typechecked
+
+from arkouda.client import generic_msg
+from arkouda.dtypes import dtype
+from arkouda.dtypes import (
+    int_scalars
+)
+from arkouda.logger import getArkoudaLogger
+
+logger = getArkoudaLogger(name="sparrayclass")
+
+class sparray:
```

Comment:
@stress-tess This is the effort that MikeO and Brad have been taking about regarding sparse matrix multiplication in Arkouda. What's going on here is that we've added a sparse matrix class to the Python side and the name chosen here is `sparray`.

I am wondering your thoughts on the naming here, are you alright with `sparray` for that? I wonder if having matrix, rather than array would be good, but naming has never been a strength of mine... 

I think the reason why this feels odd to me is even the creation functions are called like `random_sparse_matrix`, which seems it should logically return something with matrix in the name, but instead you get an array.

Reply:
The reason I decided to call it `sparray` here is a parallel to `pdarray` but it's `sp`arse.
The class also has an `ndim` field, which right now is always 2, so it is a 2D matrix, but there isn't anything from people creating more functions that create just a 1D sparse array and operate on that.

Reply:
As a side note scipy also calls their sparse arrays `sparray` (`spmatrix` for 2D), the fact that I also chose `sparray` is completely coincidental, but a nice alignment with the existing name out there.

https://docs.scipy.org/doc/scipy/reference/sparse.html#sparse-matrix-classes

Reply:
>  but a nice alignment with the existing name out there.

I wonder if we'd want to call ours something like sppdarray or pdsparray to distinguish the two and nod to Arkouda's pd ("parallel distributed") arrays—not a name I'm crazy about, but it's a precedent.

Reply:
After discussing with Shreyas, I think that we are going to merge this with the name as-is for now with the current name and then leave it open for further discussion as we continue to make improvements here.